File: packages/dotnet/src/SlyBrowser/LicenseFileReadOptions.cs
namespace SlyBrowser;

public sealed class LicenseFileReadOptions
{
    public bool AllowInsecureLocalhost { get; init; }
    public string? LicenseFilePassphrase { get; init; }
    public IReadOnlyDictionary<string, byte[]> LicenseFileTrustedKeys { get; init; } =
        new Dictionary<string, byte[]>(StringComparer.Ordinal);
    public IReadOnlySet<string> TrustedServiceUrls { get; init; } =
        new HashSet<string>(["https://api.slybrowser.com"], StringComparer.Ordinal);
}

File: packages/java/src/main/java/com/slybrowser/LicenseFileReadOptions.java
package com.slybrowser;

import java.util.Map;
import java.util.Set;

public final class LicenseFileReadOptions {
  public boolean allowInsecureLocalhost;
  public String licenseFilePassphrase;
  public Map<String, byte[]> licenseFileTrustedKeys = Map.of();
  public Set<String> trustedServiceUrls = Set.of("https://api.slybrowser.com");

  public LicenseFileReadOptions() {}

  public LicenseFileReadOptions(boolean allowInsecureLocalhost) {
    this.allowInsecureLocalhost = allowInsecureLocalhost;
  }
}
